packages/tenant-process/src/services/tenantService.ts

```diff
@@ -261,6 +288,87 @@ export function tenantServiceBuilder(
       }
     },
 
+    async addCertifiedAttribute(
+      {
+        tenantId,
+        tenantAttributeSeed,
+        organizationId,
+        correlationId,
+      }: {
+        tenantId: TenantId;
+        tenantAttributeSeed: tenantApi.CertifiedTenantAttributeSeed;
+        organizationId: TenantId;
+        correlationId: string;
+      },
+      logger: Logger
+    ): Promise<Tenant> {
+      logger.info(
+        `Add certified attribute ${tenantAttributeSeed.id} to tenant ${tenantId}`
+      );
+
+      const requesterTenant = await retrieveTenant(
+        organizationId,
+        readModelService
+      );
+
+      const certifierId = getTenantCertifierId(requesterTenant.data);
+
+      const attribute = await retrieveAttribute(
+        unsafeBrandId(tenantAttributeSeed.id),
+        readModelService
+      );
+
+      if (attribute.kind !== attributeKind.certified) {
+        throw attributeNotFound(attribute.id);
+      }
+
+      if (!attribute.origin || attribute.origin !== certifierId) {
+        throw attributeDoesNotBelongToCertifier(
+          attribute.id,
+          organizationId,
+          tenantId
+        );
+      }
+
+      const targetTenant = await retrieveTenant(tenantId, readModelService);
+
+      const tenantWithNewAttribute = await assignCertifiedAttribute({
+        targetTenant: targetTenant.data,
+        attribute,
+      });
+
+      const [updatedTenant, event] = await revaluateTenantKind({
+        tenant: tenantWithNewAttribute,
+        version: targetTenant.metadata.version,
+        readModelService,
+        correlationId,
+      });
+
+      const tenantCertifiedAttributeAssignedEvent =
+        toCreateEventTenantCertifiedAttributeAssigned(
+          targetTenant.metadata.version,
+          updatedTenant,
+          attribute.id,
+          correlationId
+        );
```

Comment:
@galales I haven't understood the intended flow. When this attribute assignment involves a tenant kind change, what's the desired event set?
- the tenant inside `TenantCertifiedAttributeAssigned` doesn't have the updated kind yet, and the update is made with the next event (TenantKindUpdated)
- the tenant inside `TenantCertifiedAttributeAssigned` already contains the updated kind, and the next event (TenantKindUpdated) is just for highlighting this, without bringing any effect to the data.

Reply:
cc @beetlecrunch